mars/metric/backends/console/console_metric.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 1999-2021 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Optional, Dict, Tuple
+
+from ..metric import Counter, Gauge, Histogram, Meter, Metric
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleMetric:
+    def __init__(
+        self, name: str, description: str = "", tag_keys: Optional[Tuple[str]] = None
+    ):
+        self._name = name
+        self._description = description
+        self._tag_keys = tag_keys
+        self._value = 0
+
+    def update(self, value: float = 1.0, tags: Optional[Dict[str, str]] = None):
+        self._value = value
+        logger.debug(
+            "Reporting metric with name: %s, description: %s, value: %s, tags: %s",
+            self._name,
+            self._description,
+            value,
+            tags,
+        )
+
+    @property
+    def value(self):
+        return self._value
+
+
+class ConsoleMetric(Metric):
+    @property
+    def value(self):
+        return self._metric.value
+
+    def _init(self):
+        self._metric = SimpleMetric(self._name, self._description, self._tag_keys)
+
+    def _record(self, value=1, tags: Optional[Dict[str, str]] = None):
+        self._metric.update(value, tags)
+
+
+class CounterImpl(ConsoleMetric, Counter):
```

Comment:
I'd like to remove `Impl` suffix of these methods, rename base classes as `AbstractXXX` and `ConsoleMetric` as `ConsoleMetricMixin`. For instance, `Counter` -> `AbstractCounter` and `CounterImpl` -> `Counter`. These names seems more pythonic. @qinxuye what's your idea?

Reply:
Reasonable to me.

Reply:
That's a good suggestion.